src/contracts/core/Slasher.sol

```diff
@@ -83,534 +26,45 @@ contract Slasher is Initializable, OwnableUpgradeable, ISlasher, Pausable {
-     *         is slashable until serveUntilBlock
-     * @param operator the operator whose stake update is being recorded
-     * @param serveUntilBlock the block until which the operator's stake at the current block is slashable
-     * @dev removes the middleware's slashing contract to the operator's linked list and revokes the middleware's (i.e. caller's) ability to
-     * slash `operator` once `serveUntilBlock` is reached
-     */
-    function recordLastStakeUpdateAndRevokeSlashingAbility(
-        address operator,
-        uint32 serveUntilBlock
-    ) external onlyRegisteredForService(operator) {
-        // update the 'stalest' stakes update time + latest 'serveUntilBlock' of the `operator`
-        _recordUpdateAndAddToMiddlewareTimes(operator, uint32(block.number), serveUntilBlock);
-        // remove the middleware from the list
-        require(
-            _operatorToWhitelistedContractsByUpdate[operator].remove(_addressToUint(msg.sender)) != 0,
-            "Slasher.recordLastStakeUpdateAndRevokeSlashingAbility: Removing middleware unsuccessful"
-        );
-        // revoke the middleware's ability to slash `operator` after `serverUntil`
-        _revokeSlashingAbility(operator, msg.sender, serveUntilBlock);
-    }
-
-    // VIEW FUNCTIONS
-
-    /// @notice Returns the block until which `serviceContract` is allowed to slash the `operator`.
-    function contractCanSlashOperatorUntilBlock(
-        address operator,
-        address serviceContract
-    ) external view returns (uint32) {
-        return _whitelistedContractDetails[operator][serviceContract].contractCanSlashOperatorUntilBlock;
-    }
-
-    /// @notice Returns the block at which the `serviceContract` last updated its view of the `operator`'s stake
-    function latestUpdateBlock(address operator, address serviceContract) external view returns (uint32) {
-        return _whitelistedContractDetails[operator][serviceContract].latestUpdateBlock;
-    }
-
-    /*
-     * @notice Returns `_whitelistedContractDetails[operator][serviceContract]`.
-     * @dev A getter function like this appears to be necessary for returning a struct from storage in struct form, rather than as a tuple.
+     * @notice External function called by AVSs to register an operator without the ability to slash it
+     * @param operator The address of the registering operator
+     * @param expiry The desired expiry time of the operator's signature
+     * @param signature The signature of the operator
      */
-    function whitelistedContractDetails(
-        address operator,
-        address serviceContract
-    ) external view returns (MiddlewareDetails memory) {
-        return _whitelistedContractDetails[operator][serviceContract];
+    function registerOperatorWithoutSlashingWithAVS(address operator, uint256 expiry, bytes memory signature) external {
+        require(expiry > block.timestamp, "Slasher.registerOperatorWithoutSlashingWithAVS: signature expired");
```

Comment:
Would prefer `registerOperatorNoSlashing` and `deregisterOperatorNoSlashing`

Reply:
Oh and `calculateRegistrationNoSlashingDigest`

Reply:
got it

Reply:
what do you think the storage should be called?

Reply:
`registeredNoSlashing`, I suppose?

Reply:
what if we just call it `registered` (or similar) and slashing-related things will live in slashing-specific storage?
I realize this is kind of getting into the future state model of slashing, but am imagining something like:
```
mapping(operator => AVS => bool) registeredForAVS
mapping(operator => AVS => slashingInfo) slashingPermissions
```
and that we limit transitions in a way where, e.g. `slashingPermissions[operator][AVS]` can only transition to non-empty if `registeredForAVS[operator][AVS] == true`

or we can make a storage struct right now, and combine the mappings? something like:
```
mapping(operator => AVS => RegistrationInfo) registrationInfo
struct RegistrationInfo {
bool operatorIsRegistered
slashingInfo (unspecified)
}
```